src/users/list_users/v2.rs

```diff
@@ -0,0 +1,107 @@
+//! [GET /_synapse/admin/v2/users/:user_id](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst#list-accountshttps://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst#query-user-account)
+
+use crate::serde::boolean_as_uint;
+use ruma::{api::ruma_api, UInt};
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "list users endpoint",
+        method: GET,
+        name: "list_users",
+        path: "/_synapse/admin/v2/users",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    #[derive(Default)]
+    request: {
+        /// Offset in the returned list. Defaults to 0.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub from: Option<UInt>,
+
+        /// Maximum amount of users to return. Defaults to 100.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub limit: Option<UInt>,
+
+        /// user_id is optional and filters to only return users with user IDs that contain this value. This parameter is ignored when using the name parameter.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub user_id: Option<String>,
+
+        /// name is optional and filters to only return users with user ID localparts or displaynames that contain this value.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub name: Option<String>,
+
+        /// The parameter guests is optional and if false will exclude guest users. Defaults to true to include guest users.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub guests: Option<bool>,
```

Comment:
Same here, except it needs to use `ruma::serde::default_true` and `ruma::serde::is_true`.